grakn-graph/src/main/java/ai/grakn/graph/internal/AbstractGraknGraph.java

```diff
@@ -958,45 +981,46 @@ private String getRelationIndex(RelationImpl relation){
+     */
+    @Override
+    public boolean duplicateResourcesExist(String index, Set<ConceptId> resourceVertexIds){
+        ResourceImpl<?> mainResource = (ResourceImpl<?>) getMainConcept(index);
+        return getDuplicates(mainResource, resourceVertexIds).size() > 0;
+    }
+
+    /**
      *
      * @param resourceVertexIds The resource vertex ids which need to be merged.
      * @return True if a commit is required.
      */
     @Override
     public boolean fixDuplicateResources(String index, Set<ConceptId> resourceVertexIds){
-        Set<ResourceImpl> duplicates = resourceVertexIds.stream()
-                .map(id -> this.<ResourceImpl>getConceptRawId(id.getValue()))
-                //filter non-null, will be null if previously deleted/merged
-                .filter(Objects::nonNull)
-                .collect(toSet());
-
-        //The "main resource" will be the one returned by the index
-        ResourceImpl<?> mainResource = getConcept(Schema.ConceptProperty.INDEX, index);
-        duplicates.remove(mainResource);
+        ResourceImpl<?> mainResource = (ResourceImpl<?>) getMainConcept(index);
+        Set<ResourceImpl> duplicates = (Set<ResourceImpl>) getDuplicates(mainResource, resourceVertexIds);
 
         //Remove any resources associated with this index that are not the main resource
-        if(duplicates.size() > 0){
-            for (ResourceImpl<?> otherResource : duplicates) {
-                Collection<Relation> otherRelations = otherResource.relations();
+        for (ResourceImpl<?> otherResource : duplicates) {
+            Collection<Relation> otherRelations = otherResource.relations();
 
-                //Copy the actual relation
-                for (Relation otherRelation : otherRelations) {
-                    copyRelation(mainResource, otherResource, (RelationImpl) otherRelation);
-                }
-
-                //Delete the node and it's castings directly so we don't accidentally delete copied relations
-                otherResource.castings().forEach(ConceptImpl::deleteNode);
-                otherResource.deleteNode();
+            //Copy the actual relation
+            for (Relation otherRelation : otherRelations) {
+                copyRelation(mainResource, otherResource, (RelationImpl) otherRelation);
             }
 
-            //Restore the index
-            String newIndex = mainResource.getIndex();
-            mainResource.getVertex().property(Schema.ConceptProperty.INDEX.name(), newIndex);
-
-            return true;
+            //Delete the node and it's castings directly so we don't accidentally delete copied relations
+            otherResource.castings().forEach(ConceptImpl::deleteNode);
+            otherResource.deleteNode();
         }
 
-        return false;
+        //Restore the index
+        String newIndex = mainResource.getIndex();
+        mainResource.getVertex().property(Schema.ConceptProperty.INDEX.name(), newIndex);
+
+        return true;
```

Comment:
Again this only returns true now. Same thing check if there are duplicates to actually merge or make it return void.